### ISSUE TITLE
Need to make GET requests with a Preview-Version header non-cacheable

### ISSUE THREAD

**Issue by magento-engcom-team** [magento/magento2#31115]
Perform a GraphQL query with a Preview-Version header to retrieve a new version of CMS page or a product.

**Expected**

Request is not cached

**Actual**

Request is being cached

**Comment by magento-engcom-team** [comment]
The issue was exported from the internal JIRA. The link to the original JIRA issue: https://jira.corp.magento.com/browse/MC-37843

**Comment by sasha19957099** [comment]
@magento I am working on this

**Comment by moloughlin** [comment]
@sasha19957099 do you mean it's being cached by Varnish? Wouldn't the default VCL inherently prohibit the caching of preview responses, since using the Preview-Version header requires you to also include an Authorization bearer token... 

And the following logic exists in vendor/magento/module-page-cache/etc/varnish5.vcl:112:

```
    # Authenticated GraphQL requests should not be cached by default
    if (req.url ~ "/graphql" && req.http.Authorization ~ "^Bearer") {
        return (pass);
    }
```

**Comment by sasha19957099** [comment]
@moloughlin Yes, it's ok for Varnish, I made it for built-in cache.

**Comment by duckchip** [comment]
is this issue fixed in 2.4.3?
